circe/core/src/main/scala/com/evolutiongaming/kafka/journal/circe/codecs.scala

```diff
@@ -0,0 +1,48 @@
+package com.evolutiongaming.kafka.journal.circe
+
+import cats.implicits._
+import com.evolutiongaming.kafka.journal.PayloadAndType._
+import com.evolutiongaming.kafka.journal.PayloadType.TextOrJson
+import com.evolutiongaming.kafka.journal._
+import io.circe._
+import io.circe.generic.semiauto._
+import play.api.libs.json.JsValue
+
+import scala.concurrent.duration._
+
+object codecs {
+
+  implicit val finiteDurationEncoder: Encoder[FiniteDuration] = Encoder.encodeString.contramap(_.toString)
+  implicit val finiteDurationDecoder: Decoder[FiniteDuration] = Decoder.decodeString.emap { str =>
+    Either.catchNonFatal(Duration(str).asInstanceOf[FiniteDuration])
+      .leftMap(_ => s"cannot parse FiniteDuration from $str")
+  }
+
+  implicit val expireAfterEncoder: Encoder[ExpireAfter] = finiteDurationEncoder.contramap(_.value)
+  implicit val expireAfterDecoder: Decoder[ExpireAfter] = finiteDurationDecoder.map(ExpireAfter(_))
+
+  implicit val jsValueEncoder: Encoder[JsValue] = Encoder.instance(convertPlayToCirce)
```

Comment:
We use it to derive Codec[PayloadMetadata]. 
PayloadMetadata still uses JsValue for its data field. Though it would be so much easier to convert JsValue to circe Json, than to extract it as a type parameter and maybe it's not necessary... (it would require second type parameter for Events[A, B], first for event's payload, second for metadata's data). or maybe there's some other way to get rid of JsValue there? What do you think?

hence all these convertCirceToPlay/convertPlayToCirce methods..

Reply:
this basically blocks idea of having one or another json lib without requiring `play-json`
I think this is something we can have for now…

Reply:
agree, but I think that this could be done later as a separate change
I've tried it and it does require much effort, cause metadata reads/writes are buried deeply in journal internals and it is not really obvious for me how to do it and I believe that it's something that can be done as a separate change